src/classes/asettings.cpp

```diff
@@ -19,23 +19,142 @@
+    {UserData::PilSortColumn,     QStringLiteral("pilSortColumn")},
+    {UserData::AcAllowIncomplete, QStringLiteral("acAllowIncomplete")},
+};
+
+QMap<ASettings::FlightLogging, QString> ASettings::flightLoggingMap = {
+    {FlightLogging::Function,           QStringLiteral("function")},
+    {FlightLogging::Approach,           QStringLiteral("approach")},
+    {FlightLogging::NightLogging,       QStringLiteral("nightlogging")},
+    {FlightLogging::LogIFR,             QStringLiteral("logIfr")},
+    {FlightLogging::FlightNumberPrefix, QStringLiteral("flightnumberPrefix")},
+    {FlightLogging::NumberTakeoffs,     QStringLiteral("numberTakeoffs")},
+    {FlightLogging::NumberLandings,     QStringLiteral("numberLandings")},
+    {FlightLogging::PopupCalendar,      QStringLiteral("popupCalendar")},
+    {FlightLogging::PilotFlying,        QStringLiteral("pilotFlying")},
+    {FlightLogging::NightAngle,         QStringLiteral("nightangle")},
+    {FlightLogging::Rules,              QStringLiteral("rules")},
+};
+
+QMap<ASettings::Setup, QString> ASettings::setupMap = {
+    {Setup::SetupComplete, QStringLiteral("setup_complete")},  // inconsistent naming
+};
+
+QMap<ASettings::NewFlight, QString> ASettings::newFlightMap = {
+    {NewFlight::FunctionComboBox, QStringLiteral("FunctionComboBox")},  // inconsistent naming
+    {NewFlight::CalendarCheckBox, QStringLiteral("calendarCheckBox")},
+};
+
 void ASettings::setup()
 {
     QSettings::setDefaultFormat(QSettings::IniFormat);
     QSettings();
 }
 
-QVariant ASettings::read(const QString &key)
-{
-    return QSettings().value(key);
-}
+//
+// Read/Write
+//
+
+QVariant ASettings::read(const FlightLogging key)
+{ return QSettings().value(pathOfKey(key)); }
+
+void ASettings::write(const FlightLogging key, const QVariant &val)
+{ QSettings().setValue(pathOfKey(key), val); }
+
+QVariant ASettings::read(const LogBook key)
+{ return QSettings().value(pathOfKey(key)); }
+
+void ASettings::write(const LogBook key, const QVariant &val)
+{ QSettings().setValue(pathOfKey(key), val); }
+
+QVariant ASettings::read(const Main key)
+{ return QSettings().value(pathOfKey(key)); }
+
+void ASettings::write(const Main key, const QVariant &val)
+{ QSettings().setValue(pathOfKey(key), val); }
+
+QVariant ASettings::read(const Setup key)
+{ return QSettings().value(pathOfKey(key)); }
+
+void ASettings::write(const Setup key, const QVariant &val)
+{ QSettings().setValue(pathOfKey(key), val); }
+
+QVariant ASettings::read(const NewFlight key)
+{ return QSettings().value(pathOfKey(key)); }
+
+void ASettings::write(const NewFlight key, const QVariant &val)
+{ QSettings().setValue(pathOfKey(key), val); }
+
+QVariant ASettings::read(const UserData key)
+{ return QSettings().value(pathOfKey(key)); }
+
+void ASettings::write(const UserData key, const QVariant &val)
+{ QSettings().setValue(pathOfKey(key), val); }
+
+//
+// QString conversion PATH
+//
+QString ASettings::pathOfKey (const ASettings::FlightLogging key)
```

Comment:
`groupOfKey`, `sectionOfKey` or something similar  would be a more suitable name. path makes it seem like a directory or file system related value, whereas it is actually just the QSettings class' way of logically [grouping together items](https://doc.qt.io/qt-5/qsettings.html#section-and-key-syntax) of the same category to read and write them under the same header in the INI file.